tests/baselines/reference/indexSignatures1.js

```diff
@@ -0,0 +1,318 @@
+//// [indexSignatures1.ts]
+// Symbol index signature checking
+
+const sym = Symbol();
+
+function gg3(x: { [key: string]: string }, y: { [key: symbol]: string }, z: { [sym]: number }) {
+    x = z;
+    y = z;  // Error
+}
+
+// Overlapping index signatures
+
+function gg1(x: { [key: `a${string}`]: string, [key: `${string}a`]: string }, y: { [key: `a${string}a`]: string }) {
+    x = y;
+    y = x;
+}
+
+interface IX { [key: `a${string}`]: string, [key: `${string}a`]: string }
+interface IY { [key: `a${string}a`]: string }
+
+function gg2(x: IX, y: IY) {
+    x = y;  // Error
+    y = x;
+}
+
+// Intersection of multiple applicable index signatures
+
+declare let combo: { [x: `foo-${string}`]: 'a' | 'b' } & { [x: `${string}-bar`]: 'b' | 'c' };
+const x1 = combo['foo-test'];  // 'a' | 'b'
+const x2 = combo['test-bar'];  // 'b' | 'c'
+const x3 = combo['foo-test-bar'];  // 'b' (('a' | 'b') & ('b' | 'c'))
+
```

Comment:
Can we also add a test for
```ts
declare var str: string;
const x4 = combo[`foo-${str}-bar`];
```
? Mostly because element access locations aren't literal locations, per sey, in `main` right now (and that was a change I had to make in my PR).
```

Reply:
Fixed in latest commit.